src/web/extractors.rs

```diff
@@ -758,15 +756,12 @@ impl FromRequest for CollectionPostRequest {
             };
 
             let max_post_records = i64::from(state.limits.max_post_records);
-            let (user_id, db, collection, query, mut bsos) =
-                <(
-                    HawkIdentifier,
-                    Box<dyn Db>,
-                    CollectionParam,
-                    BsoQueryParams,
-                    BsoBodies,
-                )>::from_request(&req, &mut payload)
-                .await?;
+
+            let user_id = HawkIdentifier::from_request(&req, &mut payload).await?;
+            let collection = CollectionParam::from_request(&req, &mut payload).await?;
+            let query = BsoQueryParams::from_request(&req, &mut payload).await?;
+            let mut bsos = BsoBodies::from_request(&req, &mut payload).await?;
```

Comment:
+100
The other is probably more efficient around stack use, but this is far more clear.

Reply:
Related: #698